algorithms/src/Kokkos_Sort.hpp

```diff
@@ -456,24 +456,37 @@ class BinSort {
   void operator()(const bin_sort_bins_tag& /*tag*/, const int i) const {
     auto bin_size = bin_count_const(i);
     if (bin_size <= 1) return;
-    int upper_bound = bin_offsets(i) + bin_size;
-    bool sorted     = false;
-    while (!sorted) {
-      sorted      = true;
-      int old_idx = sort_order(bin_offsets(i));
-      int new_idx = 0;
-      for (int k = bin_offsets(i) + 1; k < upper_bound; k++) {
-        new_idx = sort_order(k);
-
-        if (!bin_op(keys_rnd, old_idx, new_idx)) {
-          sort_order(k - 1) = new_idx;
-          sort_order(k)     = old_idx;
-          sorted            = false;
-        } else {
-          old_idx = new_idx;
+    constexpr bool use_std_sort =
+#ifdef KOKKOS_ENABLE_SERIAL
+        std::is_same_v<exec_space, Kokkos::Serial> ||
+#endif
+#ifdef KOKKOS_ENABLE_OPENMP
+        std::is_same_v<exec_space, Kokkos::OpenMP> ||
+#endif
+        false;
+    int lower_bound = bin_offsets(i);
+    int upper_bound = lower_bound + bin_size;
+    if (use_std_sort && bin_size > 10) {
+      if constexpr (use_std_sort) {
```

Comment:
Because the `use_std_sort && bin_size > 10` is not constexpr, this is an extra check to hide `std::sort` from `nvcc` as it is not `__device__` function.